demoapp/demo_app_functions.py

```diff
@@ -0,0 +1,452 @@
+"""
+This code implements text summarization, category selection and tagging bills using a single prompt
+MGL sections in 'all_bills_with_mgl.pq' files were generated using extract_mgl_section.py 
+For token size > 120K,  the code impelemtns OpenAIEmbeddings and Vectorstore to split the MGL text into chunks of len = 90K for vector embeddings
+For token size < 120K, all the documents including, bill text, MGL, chapter and section names and bill title are passed into a single prompt. 
+"""
+import streamlit as st
+import pandas as pd
+import os
+import numpy as np
+from langchain.prompts import PromptTemplate
```

Comment:
This is STILL an erroneous langchain import, right? I assume this code will not actually run for you - it should fail at import. If it DOES run, that probably means your local env is not up to date with `requirements.txt`.

Reply:
Got it, thanks for explaining!